core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -241,10 +245,55 @@ public VirtualHostBuilder tls(KeyManagerFactory keyManagerFactory,
         requireNonNull(keyManagerFactory, "keyManagerFactory");
         requireNonNull(tlsCustomizer, "tlsCustomizer");
 
-        tls(buildSslContext(() -> SslContextBuilder.forServer(keyManagerFactory), tlsCustomizer));
+        final SslContext sslContext = buildSslContext(() -> SslContextBuilder.forServer(keyManagerFactory),
+                                                      tlsCustomizer);
+        try {
+            validateKeyStoreNullPassword(sslContext);
+        } catch (Exception e) {
+            throw new RuntimeException("failed to validate SSL/TLS configuration: " + e.toString(), e);
+        }
+
+        tls(sslContext);
         return this;
     }
 
+    /**
+     * Validate key store which is used for building {@code sslContext}.
+     * If key store is initialized with null password, an exception will be raised.
+     */
+    private void validateKeyStoreNullPassword(SslContext sslContext) throws Exception {
+        SSLEngine serverEngine = null;
+        SSLEngine clientEngine = null;
+
+        try {
+            serverEngine = sslContext.newEngine(ByteBufAllocator.DEFAULT);
+            serverEngine.setUseClientMode(false);
+            serverEngine.setNeedClientAuth(false);
+
+            final SelfSignedCertificate ssc = new SelfSignedCertificate("foo.com");
+            final SslContext sslContextClient =
+                    buildSslContext(() -> SslContextBuilder.forClient()
+                                                           .keyManager(ssc.certificate(), ssc.privateKey()),
+                                    sslContextBuilder -> {});
```

Comment:
I guess we do not need to set the `KeyManagerFactory` for the client side context, because we do not perform client authentication.

Reply:
Removed.